### ISSUE TITLE
Laravel 6+ and Telescope incompatibility 

### ISSUE THREAD

**Issue by dieguiviti** [laravel/telescope#746]
- Telescope Version: v.2.1
- Laravel Version: v5.8
- PHP Version: 7.3

### Description:
I have been trying to use Laravel Telescope for a while now with laravel 6.

No specific issue or error comes up but:
-composer.json does not get modified ONLY composer.lock

meaning these do not get created:
-app/Providers/TelescopeServiceProvider.php
-config/telescope.php
-public/vendor/

### Steps To Reproduce:
I manually changed the version of laravel/framework in composer.json to 5.8, 
ran composer update
and managed to install telescope successfully

THE ISSUE IS CLEARLY AN INCOMPATIBILITY BETWEEN LARAVEL 6 AND TELESCOPE

**Comment by billriess** [comment]
Strange, we're having no issues running Telescope on Laravel 6.

Are you sure your composer file has the right versions? What version of telescope are you running?

**Comment by lk77** [comment]
Hello,

there is some issues with Laravel 6 currently, 
installation fails like https://github.com/laravel/telescope/issues/732,
and there is some issues with csrf token also.

**Comment by ludo237** [comment]
Yes it is

**Comment by didieruzan** [comment]
Same problem here with Laravel 6.6.0 and Telescope

**Comment by taylorotwell** [comment]
I can install it fine on the latest release of Laravel 6.

**Comment by Ghosstk** [comment]
Hey!

I cant install telescope with a fresh laravel install what I just made today.
Laravel: "v6.6.2"
Telescope "v2.1.1"

Telescope is in my json.lock after the "install".
But its not working. No provider, config, migration or anything.

Install:
$php artisan telescope:install

Publishing Telescope Service Provider...
Publishing Telescope Assets...
Publishing Telescope Configuration...
Telescope scaffolding installed successfully.

Migrate:
$php artisan migrate

Nothing to migrate.

Publish:
$php artisan telescope:publish

Unable to locate publishable resources.
Publishing complete.
Unable to locate publishable resources.
Publishing complete.

**Comment by indigoram89** [comment]
@Ghosstk  I have the same error too

**Comment by lk77** [comment]
i still have those error too,
i can't install telescope, i need to copy the stub files manually.

**Comment by mullervanas** [comment]
I have the same issue with Laravel 6 also on Windows 10. Will there be a fix for this as I can't seem to get it working copying the files over manually.

**Comment by joostvanhoof** [comment]
Having exactly the same issue here as @Ghosstk describes. Service Provider, assets, config and migration are nowhere to be found.

**PHP:** 7.4
**OS:** MacOS
**Laravel:** 6.12.0
**Telescope:** 2.1

First time I ran `php artisan telescope:install` I got the error below (full exception trace at the bottom of this post):
```
file_get_contents(/Users/joost/Documents/dev/tentacle/app/Providers/TelescopeServiceProvider.php): failed to open stream: No such file or directory
```

Then ran it again and got:
```
php artisan telescope:install
Publishing Telescope Service Provider...
Publishing Telescope Assets...
Publishing Telescope Configuration...
Telescope scaffolding installed successfully.
```

But the scaffolding is not there. @indigoram89, @lk77, @MVA18 has anyone of you managed to fix this?

Full exception trace:
```
ErrorException  : file_get_contents(/Users/joost/Documents/dev/tentacle/app/Providers/TelescopeServiceProvider.php): failed to open stream: No such file or directory

  at /Users/joost/Documents/dev/tentacle/vendor/laravel/telescope/src/Console/InstallCommand.php:80
    76|
    77|         file_put_contents(app_path('Providers/TelescopeServiceProvider.php'), str_replace(
    78|             "namespace App\Providers;",
    79|             "namespace {$namespace}\Providers;",
  > 80|             file_get_contents(app_path('Providers/TelescopeServiceProvider.php'))
    81|         ));
    82|     }
    83| }
    84|

  Exception trace:

  1   file_get_contents("/Users/joost/Documents/dev/tentacle/app/Providers/TelescopeServiceProvider.php")
      /Users/joost/Documents/dev/tentacle/vendor/laravel/telescope/src/Console/InstallCommand.php:80

  2   Laravel\Telescope\Console\InstallCommand::registerTelescopeServiceProvider()
      /Users/joost/Documents/dev/tentacle/vendor/laravel/telescope/src/Console/InstallCommand.php:43
```

**Comment by dellow** [comment]
Also having exactly the same issue as @Ghosstk. Have tried updating dependencies and still have the same issue.

**Comment by DmitriyKashin** [comment]
Same issue.

**Comment by AidynMakhataev** [comment]
same issue with last laravel version

**Comment by dasraab** [comment]
yep, me too

**Comment by barryvdh** [comment]
Do you get any errors when running 'php artisan vendor:publish --tag=telescope-provider' directly? Does the ServiceProvider actually exists at that location?

**Comment by renatop7** [comment]
> Do you get any errors when running 'php artisan vendor:publish --tag=telescope-provider' directly? Does the ServiceProvider actually exists at that location?

Hi @barryvdh 
I'm having the exactly same issue as @Ghosstk and @joostvanhoof 

When I run `php artisan vendor:publish --tag=telescope-provider` I get an error message:
```
Unable to locate publishable resources.
Publishing complete.
```

**Comment by retax3** [comment]
I tried this series of commands and it work for me
composer require laravel/telescope 

change in composer.json version of telescope to 2.1

composer install

php artisan telescope:install

php artisan migrate

php artisan telescope:publish

go to route /telescope 

I hope it works for you !! ;)

**Comment by joostvanhoof** [comment]
Telescope is currently at version 3.1 so this would just postpone the problem.

**Comment by lk77** [comment]
The only solution i've found was to copy the files manually.

**Comment by iRynoh** [comment]
> The only solution i've found was to copy the files manually.

What did you copy, the public folder?

**Comment by ttawan** [comment]
Following.

**Comment by metrixinfo** [comment]
Same issue here on a Mac.

Fresh installs of Laravel and Telescope.

**Comment by jacobshenning** [comment]
I can confirm this issue. Temporary solution is to just copy the stub manually.

- [ ] Locate the stub - `~project/vendor/laravel/telescope/stubs/TelescopeServiceProvider.stub`
- [ ] Copy the stub into `~project/app/providers/`
- [ ] Rename the stub from _TelescopeServiceProvider.stub_ to _TelescopeServiceProvider.php_

Or if you're on a mac, just go to your project folder in terminal and run this.

`cp vendor/laravel/telescope/stubs/TelescopeServiceProvider.stub app/providers/TelescopeServiceProvider.php`

Is there a chance this is a permissions error?

**Comment by wireblue** [comment]
I think my issue is slightly different to the OP, but I was getting the same `"Unable to locate publishable resources."` error when trying to run the `artisan vendor:publish --tag=telescope-migrations` command.

Setting `TELESCOPE_ENABLED=true` in my `.env` solved it. Hopefully it might help some of these other scenarios also.

Note: In `Laravel\Telescope\TelescopeServiceProvider->boot()` there's an IF statement which skips boot if Telescope is not enabled. Telescope has to be booted before the migrations, config etc becomes available.

**Comment by themsaid** [comment]
Please guys open a new issue with laravel 7.x installed.

**Comment by bondas83** [comment]
TELESCOPE_ENABLED=true fix a problem, because from telescope 2.1 added this line in boot:

`if (! config('telescope.enabled')) {
            return;
        }`
